src/main/kotlin/uk/gov/dwp/dataworks/dks/services/impl/HttpKeyService.kt

```diff
@@ -0,0 +1,111 @@
+package uk.gov.dwp.dataworks.dks.services.impl
+
+import com.google.gson.Gson
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.ContentType
+import org.apache.http.entity.StringEntity
+import org.apache.http.util.EntityUtils
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import uk.gov.dwp.dataworks.dks.domain.DataKeyResult
+import uk.gov.dwp.dataworks.dks.exceptions.DataKeyDecryptionException
+import uk.gov.dwp.dataworks.dks.exceptions.DataKeyServiceUnavailableException
+import uk.gov.dwp.dataworks.dks.providers.HttpClientProvider
+import uk.gov.dwp.dataworks.dks.services.KeyService
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.URLEncoder
+
+open class HttpKeyService(private val httpClientProvider: HttpClientProvider, val dataKeyServiceUrl: String) : KeyService {
+
+    companion object {
+        val logger: Logger = LoggerFactory.getLogger(HttpKeyService::class.toString())
+    }
+
+    override fun batchDataKey(): DataKeyResult {
+        try {
+            val dksUrl = "$dataKeyServiceUrl/datakey"
+            logger.info("dataKeyServiceUrl: '$dksUrl'.")
```

Comment:
Maybe this should be debug? might get spammy if this happens every key request. I think the same with a lot of the logs in the decryption key calls